adr/component_offerings.md

```diff
@@ -0,0 +1,134 @@
+
+```jsx
+<ClayCard>
+	<ClayCard.Title onClick={handleTitleClick}>
+		<ClayIcon symbol="check" />
+
+		{'Hello World'}
+	</ClayCard.Title>
+
+	<ClayCard.Body onClick={handleBodyClick}>
+		<ClayCard.Description>
+			{'Some cool descriptors'}
+		</ClayCard.Description>
+	</ClayCard.Body>
+</ClayCard>
+
+// vs
+<ClayCard
+	iconTitleLeftSymbol="check"
+	onTitleClick={handleTitleClick}
+	onBodyClick={handleBodyClick}
+	description="foo bar baz"
+	title="hello world"
+/>
+```
+
+## Decision
+
+The decision going forward is to create 2 sets of components
+
+-   low-level composable components such as `<ClayCard.Header>`, `<ClayCard.Footer>`
+-   simple high-level components that cover basic lexicon uses
+
+### What does this look like?
+
+#### Composable:
+
+![Composable Drop Down](./images/component_offerings_2.png)
+
+```jsx
+<ClayDropDown>
+	<ClayDropDown.Search
+		onChange={() => {}}
+		placeholder="Search"
+		spritemap={spritemap}
+		value={''}
+	/>
+
+	<ClayDropDown.ItemList>
+		{items.map((item, i) => (
+			<ClayDropDown.Item href={item.href} key={i}>
+				{item.label}
+			</ClayDropDown.Item>
+		))}
+	</ClayDropDown.ItemList>
+</ClayDropDown>
+```
+
+Composing gives the user more control over the layout and interactions.
+
+#### High-level:
+
+![High Level Drop Down](./images/component_offerings_1.png)
+
+```jsx
+<ClayDropDownWithPagination
+	delta={delta}
+	header="Favorites"
+	items={items} // array of objects with `label` and `href`
+	message={'Showing {0} of {1} Structures'}
+	title="Folder"
+/>
+
+// Compare if this component was composed of low-level components
+<ClayDropDown>
+	<ClayDropDown.Header header="Folder" />
+
+	<ClayDropDown.ItemList>
+		<ClayDropDown.Group header="Favorites">
+			{items.slice(0, page * delta).map((item, i) => (
+				<ClayDropDown.Item href={item.href} key={i}>
+					{item.label}
+				</ClayDropDown.Item>
+			))}
+		</ClayDropDown.Group>
+	</ClayDropDown.ItemList>
+
+	<ClayDropDown.Caption>
+		{`Showing ${page * delta} of ${items.length} structures`}
+	</ClayDropDown.Caption>
+
+	<ClayDropDown.Action onClick={() => setPage(page + 1)}>
+		{'Show More'}
+	</ClayDropDown.Action>
+</ClayDropDown>
+```
+
+You'll notice that the high-level component handles much of the interactions internally. It will handle the click of the `show more`, paginate, and update language as necessary. In fact, even this high-level component will be made from the smaller low-level building blocks.
+
+Finally, both the high-level and low-level components will be exported from each package. The low-level component will be exported as default, but the high-level ones will be exported by name. The high-level components should be put into a
+
+For example:
+
+```jsx
+export {ClayDropDownWithPagination};
+
+export default ClayDropDown;
+```
```

Comment:
If it was an option, I'd avoid using default exports ... 
If I remember correctly we've talked about this several times and it might be a good moment to think about it

https://basarat.gitbooks.io/typescript/docs/tips/defaultIsBad.html
https://humanwhocodes.com/blog/2019/01/stop-using-default-exports-javascript-module/
https://blog.neufund.org/why-we-have-banned-default-exports-and-you-should-do-the-same-d51fdc2cf2ad

I'm not saying I agree 100% with everything that's said in the URLs linked above, but I think we should talk about this just to know what we want to go with.

Reply:
> If I remember correctly we've talked about this several times

Yep, that I am aware of least here:

- https://github.com/liferay/clay/issues/1495
- https://github.com/liferay/clay/pull/1453

Reply:
If we banish standard export, we will have to deal with component naming to differentiate a high-level component for a low-level component (I am still assuming we will have these two for each component ClayDropdown, Card...), I not think of good names for differentiate them.

I agree on not using standard export the benefits are numerous but I still see this problem we will have to deal with here.

Reply:
> If we banish standard export

I don't actually wan't to "banish" them. There'll still probably be some situations in which we'll  need or want to use `export default` ... 

But as a norm/standard, I'd rather do this: 

```javascript
export ClayDropDown;
// ...
export ClayDropDownWithPagination;
```

Than this:

```javascript
export {ClayDropDownWithPagination};
export default ClayDropDown;
```

Once again, I'm just giving my opinion.

> we will have to deal with component naming to differentiate a high-level component for a low-level component 

I agree, naming is (and will be hard) whether we use `export default` or not ;)

Reply:
I don't really have any strong opinions on this, `default` for the low-level just felt right at the time 🤷‍♂